e2etests/all_test.go

```diff
@@ -0,0 +1,144 @@
+// +build e2etests
+
+package e2etests
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/gobuffalo/envy"
+	"github.com/stretchr/testify/suite"
+)
+
+type E2eSuite struct {
+	suite.Suite
+	goBinaryPath   string
+	env            []string
+	tmpDir         string
+	sampleRepoPath string
+	stopAthens     context.CancelFunc
+}
+
+type catalogRes struct {
+	Modules []struct {
+		Module  string `json:"module"`
+		Version string `json:"version"`
+	} `json:"modules"`
+}
+
+func (m *E2eSuite) SetupSuite() {
+	var err error
+	m.tmpDir, err = ioutil.TempDir("/tmp", "gopath")
+	if err != nil {
+		m.Fail("Failed to make temp dir", err)
+	}
+
+	m.sampleRepoPath, err = ioutil.TempDir("/tmp", "repopath")
+	if err != nil {
+		m.Fail("Failed to make temp dir for sample repo", err)
+	}
+
+	m.goBinaryPath = envy.Get("GO_BINARY_PATH", "go")
+
+	athensBin, err := buildAthens(m.goBinaryPath, m.tmpDir, m.env)
+	if err != nil {
+		m.Fail("Failed to build athens ", err)
+	}
+	stopAthens() // in case a dangling instance was around.
+	// ignoring error as if no athens is running it fails.
+
+	ctx := context.Background()
+	ctx, m.stopAthens = context.WithCancel(ctx)
```

Comment:
Form the context package docs: 
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter. 

I realize storing the context is necessary here because of testify/suite, but I'd rather we followed best practices instead of having the framework steer us away form that. 

That said, we can definitely do it in a follow up PR because it means removing the test suite from the e2e tests :)

Reply:
Ah right. That would make things less easy. Not even sure if it's worth reverting to the previous `stopAthens`, as even if we remove testify having a propagated context will bloat the code a bit.